api/src/pcapi/core/external/attributes/api.py

```diff
@@ -229,33 +253,74 @@ def get_bookings_categories_and_subcategories(
+            bookings_by_categories[booking.stock.offer.subcategory.category.id].append(booking)
             bookings_by_subcategories[booking.stock.offer.subcategoryId].append(booking)
 
-    sorted_subcategories_items = sorted(
-        bookings_by_subcategories.items(),
-        key=lambda key_value: (
-            len(key_value[1]),
-            sum(booking_.stock.price for booking_ in key_value[1]),
-        ),
-        reverse=True,
-    )
+    most_booked_category = _get_most_booked(bookings_by_categories)
+    most_booked_subcategory = _get_most_booked(bookings_by_subcategories)
+
+    most_booked_movie_genre = None
+    if most_booked_category == categories.CINEMA.id:
+        cinema_bookings_by_genre = defaultdict(list)
+        for booking in bookings_by_categories[categories.CINEMA.id]:
+            if extra_data := booking.stock.offer.extraData:
+                for genre in extra_data.get("genres") or []:
+                    cinema_bookings_by_genre[genre].append(booking)
+        most_booked_movie_genre = _get_most_booked(cinema_bookings_by_genre)
+
+    most_booked_music_type = None
+    music_categories = (categories.MUSIQUE_ENREGISTREE.id, categories.MUSIQUE_LIVE.id)
+    if most_booked_category in music_categories:
+        music_bookings_by_type = defaultdict(list)
+        for category in music_categories:
+            for booking in bookings_by_categories[category]:
+                if extra_data := booking.stock.offer.extraData:
+                    if music_type := extra_data.get("musicType"):
+                        music_bookings_by_type[music_type].append(booking)
+        most_booked_music_type = _get_most_booked(music_bookings_by_type)
 
-    booking_subcategories_ids = [subcategory[0] for subcategory in sorted_subcategories_items]
-    booking_categories_ids = list(booking_categories)
-    most_booked_subcategory = sorted_subcategories_items[0][0] if sorted_subcategories_items else None
     return models.BookingsAttributes(
-        booking_categories=booking_categories_ids,
-        booking_subcategories=booking_subcategories_ids,
+        # sorted alphabetically to be deterministic and avoid useless changes in external data
+        booking_categories=sorted(bookings_by_categories.keys()),
+        booking_subcategories=sorted(bookings_by_subcategories.keys()),
         most_booked_subcategory=most_booked_subcategory,
+        most_booked_movie_genre=most_booked_movie_genre,
+        most_booked_music_type=most_booked_music_type,
     )
 
 
+# Specific for "Retro" campaign in December 2022 - may become deprecated in 2023
+def get_booking_attributes_2022(
+    user_bookings: list[bookings_models.Booking],
+) -> tuple[Decimal, str | None, str | None]:
+    bookings_2022 = sorted(
+        filter(
+            lambda booking: datetime.combine(date(2022, 1, 1), datetime.min.time())
+            <= booking.dateCreated
+            <= datetime.combine(date(2022, 12, 31), datetime.max.time())
```

Comment:
(détail / style) Je pense qu'on peut écrire plus simplement : 

```suggestion
            lambda booking: booking.dateCreated.year == 2022
```

(Bon, on a un décalage d'1 heure car les dates en bases sont en UTC, mais je pense qu'on peut ignorer.)

Reply:
Bonne idée, c'est en effet plus simple, et plus lisible, merci :)
Ça ne me semble pas dramatique pour l'heure de décalage, surtout que les réservations autour de minuit la nuit du nouvel an ne doivent pas être les plus nombreuses.